nucypher/blockchain/eth/agents.py

```diff
@@ -100,13 +100,13 @@ def __init__(self,
                  transaction_gas: Optional[Wei] = None):
 
         self.log = Logger(self.__class__.__name__)
-        self.registry = registry
+        self.registry_repr = str(registry)
```

Comment:
Heh - slightly confusion name here - since `str` calls `__str__` and not `_repr__`.

Reply:
you're right
changed 
✔️